### PR TITLE
web server question enhancements

### PR DESCRIPTION
**Pull request by MichaIng**
**By submitting this pull request, I confirm the following:** 
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

**Signed-off-by: MichaIng <micha@dietpi.com>**

---
**What does this PR aim to accomplish?:**
- When the web interface is not wanted, the web server install is obsolete, hence the question to install it not required.
- When the web server is deselected during install, required PHP modules are not installed as well and the web server user is not added to the `pihole` group for database access through web UI. This needs to be instead done/assured manually by the user.

**How does this PR accomplish the above?:**
+ Do not ask to install the web server, if the web interface has been deselected before.
+ Add additional info, that PHP modules need to be installed manually, when web server is deselected and the web server user needs to be member of the "pihole" group.

### CODE REVIEW

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/error-message-attempt-to-write-a-readonly-database/27092/29

File: automated install/basic-install.sh
Comment (DL6ER) [inline automated install/basic-install.sh:1183]: Can we have the `PIHOLE_WEB_DEPS` be the source for the packages to get installed? Otherwise, when will almost certainly forget to update this bit when we have to add another dependency in, say, six months.

https://github.com/pi-hole/pi-hole/blob/d42785a3bfa793f9fa494096b27b6b7b72b0bbe3/automated%20install/basic-install.sh#L250

Comment (MichaIng) [reply automated install/basic-install.sh:1183]: @DL6ER 
Good idea:
```
$(for i in $PIHOLE_WEB_DEPS; do [[ $i == php* && $i != *-cgi ]] && echo -n "${i#*-} "; done)
```
- Not beautiful but we want the "php[X.Y]-" prefix and cgi not included, the latter since fpm or mod-php or other implementations would do as well.
- PHP json is missing on Debian then, since it is not in the list, most likely since it is dependency of all PHP handler packages, fpm, cgi and mod-php, hence pulled anyway. This is probably okay. We could add this one manually, but then needed to check for apt vs yum, otherwise it can be doubled. Or we could add it to the array on Ubuntu/Debian as well, also to be failsafe since it is definitely required?

Comment (MichaIng) [reply automated install/basic-install.sh]: @DL6ER 
Okay done. I also shortened the output a bid so that it fits into a minimal whiptail:
pihole

**Comment by MichaIng** [comment]
@DL6ER 
Conflicts have been resolved and your requested changes have been added. What do you think?
Considering the amount of help requests in the forum around Nginx and Apache webserver systems, I think the additional information would be great to have with next release 🙂.

**Review by PromoFaux — APPROVED** [review]

**Comment by PromoFaux** [comment]
I'll wait for approval from @DL6ER to merge this, though. As I see he requested changes

**Review by DL6ER — APPROVED** [review]

**Pull request merged**
